packages/core/src/dialects/cockroachdb/query.js

```diff
@@ -0,0 +1,159 @@
+import { AbstractQuery } from '../abstract/query';
+import _ from 'lodash';
+import logger from '../../utils/logger';
+import sequelizeErrors from '../../errors';
+import { defaults, flattenObjectDeep, cloneDeep } from '../../utils/object';
+import { intersects } from '../../utils/array';
+
+export class CockroachDbQuery extends AbstractQuery {
+  constructor() {
+    super();
+
+    this.model.findOrCreate = async options => {
```

Comment:
You can't modify a model method dynamically like this. Instead, I recommend modifying findOrCreate and findByPk in model.js to support your use case